tool/tsh/common/tsh_helper_test.go

```diff
@@ -413,3 +416,58 @@ func mustCloneTempDir(t *testing.T, srcDir string) string {
 	require.NoError(t, err)
 	return dstDir
 }
+
+func mustMakeDynamicKubeCluster(t *testing.T, name, discoveredName string, labels map[string]string) types.KubeCluster {
```

Comment:
hmm. i think `types.DiscoveredNameLabel` should be limited to only cloud resources. I don't know if it's a security risk though so we may not care to restrict this on the auth side. But at least for testing, use `types.OriginCloud`?